rdrf/rdrf/forms/dynamic/fields.py

```diff
@@ -44,17 +45,23 @@ def validate(self, value):
 
 class FileTypeRestrictedFileField(FileField):
 
+    def _find_whitelisted_file_types(self, file_extension):
+        return WhitelistedFileExtension.objects.filter(file_extension__iexact=file_extension)
+
     def _find_blacklisted_mime_type(self, mt):
         from rdrf.models.definition.models import BlacklistedMimeType
         return BlacklistedMimeType.objects.filter(mime_type=mt).first()
 
     def validate(self, value):
         if not value:
             return super().validate(value)
-        __, ext = os.path.splitext(value._name)
+        __, ext = os.path.splitext(value.name)
         mime_type = magic.from_buffer(value.file.read(2048), mime=True)
         value.file.seek(0)
 
+        if not self._find_whitelisted_file_types(ext):
+            raise ValidationError(_(f"{ext} is not a supported file extension."))  # contact details if they believe it should be supported?
```

Comment:
Yes to the comment. Perhaps we need to add curator details as a field to the `Registry` object?

Reply:
Yes, I think that's a great idea.

Reply:
The file field `__init__` method is called when the application starts up, so I am not sure if I can pass in the Registry object to get the `contact_email` from it.

Any thoughts?

Reply:
We could say refer to the footer, and make sure we update that in all the registries.

Reply:
A new setting to capture the CURATOR_EMAIL has been created to support this. The registry repos will need to be updated with relevant contacts for mnd, arrk, angelman.